cmd/parse/parse.go

```diff
@@ -153,25 +152,35 @@ func enqueueMissingBlocks(exportQueue types.HeightQueue, ctx *Context) {
 	}
 }
 
-// startNewBlockListener subscribes to new block events via the Tendermint RPCConfig
-// and enqueues each new block height onto the provided queue. It blocks as new
-// blocks are incoming.
-func startNewBlockListener(exportQueue types.HeightQueue, ctx *Context) {
-	eventCh, cancel, err := ctx.Node.SubscribeNewBlocks("juno-new-blocks-listener")
-	defer cancel()
-
+// enqueueNewBlocks enqueues new block heights onto the provided queue.
+func enqueueNewBlocks(exportQueue types.HeightQueue, ctx *Context) {
```

Comment:
I think this function can be simplified as follows: 
```go
// enqueueNewBlocks enqueues new block heights onto the provided queue.
func enqueueNewBlocks(exportQueue types.HeightQueue, ctx *Context) {
	currHeight, err := ctx.Node.LatestHeight()
	if err != nil {
		panic(fmt.Errorf("failed to get last block from RPCConfig client: %s", err))
	}
	
	for {
		// Get the latest block height
		latestBlockHeight, err := ctx.Node.LatestHeight()
		if err != nil {
			panic(fmt.Errorf("failed to get last block from RPCConfig client: %s", err))
		}

		// Enqueue all heights from the current height up to the latest height
		for ; currHeight <= latestBlockHeight; currHeight++ {
			ctx.Logger.Debug("enqueueing new block", "height", currHeight)
			exportQueue <- currHeight
		}

		time.Sleep(config.Cfg.Parser.AvgBlockTime)
	}
}
```

What are your thoughts here?

Reply:
HA thats true. let me push the update